src/page-view.ts

```diff
@@ -0,0 +1,162 @@
+import { uuidv7 } from './uuidv7'
+import * as utils from './utils'
+
+interface PageViewData {
+    // scroll is how far down the page the user has scrolled,
+    // content is how far down the page the user can view content
+    // (e.g. if the page is 1000 tall, but the user's screen is only 500 tall,
+    // and they don't scroll at all, then scroll is 0 and content is 500)
+    maxScrollHeight?: number
+    maxScrollY?: number
+    lastScrollY?: number
+    maxContentHeight?: number
+    maxContentY?: number
+    lastContentY?: number
+}
+
+interface ScrollProperties {
+    $prev_pageview_last_scroll?: number
+    $prev_pageview_last_scroll_percentage?: number
+    $prev_pageview_max_scroll?: number
+    $prev_pageview_max_scroll_percentage?: number
+    $prev_pageview_last_content?: number
+    $prev_pageview_last_content_percentage?: number
+    $prev_pageview_max_content?: number
+    $prev_pageview_max_content_percentage?: number
+}
+
+interface NonPageEventProperties {
+    $pageview_id: string
+}
```

Comment:
```suggestion
```

Reply:
Damn, I added this https://github.com/PostHog/posthog-js/pull/775 so that we can't miss things like this in the future